src/material/tooltip/tooltip.ts

```diff
@@ -205,7 +205,7 @@ export class MatTooltip implements OnDestroy, AfterViewInit {
   /** The message to be displayed in the tooltip */
   @Input('matTooltip')
   get message() { return this._message; }
-  set message(value: string) {
+  set message(value: any) {
```

Comment:
I don't think that we should sacrifice type safety to appease Closure. Can we try one of these solutions to stop it from being optimized away (imagine the quotes are backticks)?

1. `"${(value as any)}".trim()`
2. `"${value} ".trim()`. Note the extra whitespace at the end which will be removed by `trim`.

Reply:
Thanks, just gave that a go. First option:

```
function x(value: string) {
  return `${(value as any)}`.trim();
}
console.log(x(document.domain));
```

still elides the stringification:

```
console.log(document.domain.trim());
```

Reply:
Second option works:
```
function x(value: string) {
  return `${value} `.trim();
}
console.log(x(document.domain));
```
compiles to
```
console.log((document.domain+" ").trim());
```

Reply:
Another option (I think this is my favourite) that works:

```
function x(value: string) {
  return String(value).trim();
}
console.log(x(document.domain));
```
compiles to
```
console.log(String(document.domain).trim());
```

Reply:
I like the third option as well, just make sure to add a comment about why we have to write it this way.

Reply:
Done, thanks for your thoughtful review -- this is much better :-)